modules/core/src/lib/layer.js

```diff
@@ -244,6 +243,19 @@ export default class Layer extends Component {
     return screenPixels * devicePixelRatio;
   }
 
+  // Event handling
+  onHover(info) {
+    if (this.props.onHover) {
+      this.props.onHover(info);
+    }
+  }
+
+  onClick(info) {
+    if (this.props.onClick) {
+      this.props.onClick(info);
+    }
+  }
+
```

Comment:
how about onDrag?

Reply:
onDragStart, onDrag, onDragEnd

Reply:
See [deck.js:617](https://github.com/uber/deck.gl/pull/2427/files#diff-2c3ba220c52980e05b7f0f83e67bb7e5R617) - The new callbacks are added dynamically so they don't have default layer implementations.